### PR TITLE
Add PointerScaleEvent and use in InteractiveViewer

### PR DESCRIPTION
**Pull request by moffatman**
Add new type of `PointerSignal`, and handle it within `InteractiveViewer`. For use in trackpad pinch-to-zoom where full gesture information is unavailable (web).

Part of #112103

## Sequence

1. https://github.com/flutter/flutter/pull/112170
2. https://github.com/flutter/engine/pull/36342
3. This PR
4. https://github.com/flutter/engine/pull/36348

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -1917,6 +1917,103 @@ class _TransformedPointerScrollInertiaCancelEvent extends _TransformedPointerEve
...
+  PointerScaleEvent copyWith({
+    Duration? timeStamp,
+    int? pointer,
+    PointerDeviceKind? kind,
+    int? device,
+    Offset? position,
+    Offset? delta,
+    int? buttons,
+    bool? obscured,
+    double? pressure,
```
Comment (justinmc) [inline packages/flutter/lib/src/gestures/events.dart:1934]: Why are some of these parameters not used?

Comment (justinmc) [reply packages/flutter/lib/src/gestures/events.dart:1934]: I guess PointerPanZoomUpdateEvent.copyWith is doing the same thing below.

File: packages/flutter/lib/src/widgets/interactive_viewer.dart [hunk 2]
```diff
@@ -951,55 +951,62 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
 
   // Handle mousewheel scroll events.
   void _receivedPointerSignal(PointerSignalEvent event) {
+    double scaleChange = 1;
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/interactive_viewer.dart:954]: Are you still able to use `final` here?  I always forget when that works or not...

Or `late final` if not.

File: packages/flutter/test/widgets/interactive_viewer_test.dart [hunk 3]
```diff
@@ -1705,6 +1705,38 @@ void main() {
...
+      final TestPointer pointer = TestPointer(1, PointerDeviceKind.trackpad);
+      await tester.sendEventToBinding(pointer.hover(tester.getCenter(find.byType(SizedBox))));
+      await tester.sendEventToBinding(pointer.scale(1.5));
+      await tester.pump();
+      expect(transformationController.value.getMaxScaleOnAxis(), 1.5);
+
+      // Send another scale event.
+      await tester.sendEventToBinding(pointer.scale(1.5));
+      await tester.pump();
+      expect(transformationController.value.getMaxScaleOnAxis(), 2.25);
```
Comment (justinmc) [inline packages/flutter/test/widgets/interactive_viewer_test.dart:1738]: Nit: Maybe also show that this won't exceed maxScale (2.5) if you think that would be useful?

**Review by justinmc — COMMENTED** [review]
Thanks for all the InteractiveViewer improvements.  This seems reasonable to me.  I'll review the engine PR and then come back to this when it's merged.  I want to try this on my machine locally once that's merged too.

File: packages/flutter/lib/src/gestures/events.dart [hunk 4]
```diff
@@ -1917,6 +1917,103 @@ class _TransformedPointerScrollInertiaCancelEvent extends _TransformedPointerEve
...
+      embedderId: embedderId ?? this.embedderId,
+      scale: scale ?? this.scale,
+    ).transformed(transform);
+  }
+}
+
+/// The pointer issued a scale event.
+///
+/// Pinching-to-zoom in the browser is an example of an event
+/// that would create a [PointerScrollInertiaCancelEvent].
```
[the remark this answers is not in the document]
Comment (moffatman) [reply packages/flutter/lib/src/gestures/events.dart:1964]: typo yeah

File: packages/flutter/lib/src/gestures/events.dart (hunk 1 above)
Comment (moffatman) [reply packages/flutter/lib/src/gestures/events.dart:1934]: The signature has to match `PointerEvent.copyWith`, but most parameters are not relevant

**Review by justinmc — APPROVED** [review]
LGTM :+1:

**Pull request merged**